src/Keboola/GoogleSheetsWriter/Sheet.php

```diff
@@ -38,36 +39,31 @@ public function process(array $sheet): array
             $sheetProperties = $this->getSheetProperties($sheet['fileId'], $sheet['sheetId']);
             $columnCount = $this->inputTable->getColumnCount();
             $rowCountSrc = $this->inputTable->getRowCount();
-
-            if (empty($sheetProperties)) {
-                throw new UserException(sprintf(
-                    'Sheet "%s" (%s) not found in file "%s" (%s)',
-                    $sheet['sheetTitle'],
-                    $sheet['sheetId'],
-                    $sheet['title'],
-                    $sheet['fileId']
-                ));
-            }
-
-            if ($columnCount * $rowCountSrc > 2000000) {
-                throw new UserException('CSV file exceeds the limit of 2000000 cells');
-            }
+            $this->preFlightChecks($sheet, $sheetProperties, $columnCount, $rowCountSrc);
 
             // update columns
             $rowCountDst = $sheetProperties['properties']['gridProperties']['rowCount'];
             $this->updateMetadata($sheet, ['columnCount' => $columnCount, 'rowCount' => $rowCountDst]);
 
-            if ($sheet['action'] === ConfigDefinition::ACTION_UPDATE) {
-                $this->client->clearSpreadsheetValues(
-                    $sheet['fileId'],
-                    $this->getRange($sheet['sheetTitle'], $columnCount, 1, $rowCountDst)
-                );
-                // update rows to match source size
-                $this->updateMetadata($sheet, ['columnCount' => $columnCount, 'rowCount' => $rowCountSrc]);
-            }
-
             // upload data
-            return $this->uploadValues($sheet, $this->inputTable);
+            switch ($sheet['action']) {
+                case ConfigDefinition::ACTION_UPDATE:
+                    $this->client->clearSpreadsheetValues(
```

Comment:
Proč dává smysl tohle mít oddělené od té "updateAction"? Dává smysl volat updateAction bez toho, abych to vyčistil a updatnul metadata?

Reply:
Tiez by som to rad strcil do "updateAction", ale ta metoda len uploaduje data, nijak uz nepracuje s "columnCount" a "rowCount", takze by som jej musel pridat tieto argumenty.